### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by s2hc-johan**
* New structure
* tried to go through and update accordingly to current state. (not term, since so much more is coming)

### CODE REVIEW

File: docs-src/source/data_types.rst [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
 =====================
 
+Things to be aware about
+------------------------
+
+As we started this we've seen that there are some things that doesn't map
+that good between python and erlang, and that the best solution for it could
+be different pending on the problem you're currently trying to solve.
+
+Erlang strings is more of syntactic sugar, but it is infact a list of
```
Comment (kvakvs) [inline docs-src/source/data_types.rst]: Erlang strings are lists of `char()` where `char()` is Unicode symbol range from like 1 to 0x1fsomething000, 0..255 are ASCII strings which match Erlang strings too. Erlang program can send you a Unicode string which `term` will encode as a list of small and large integers and it is responsibility of Pyrlang user to know whether that is a unicode string or a list of integers.

File: docs-src/source/modules/gen/index.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
+gen module - Generic OTP-style calls
+====================================
+
+The gen module implements Erlang style behaviours ``gen_server``, and maybe
```
Comment (kvakvs) [inline docs-src/source/modules/gen/index.rst]: Clarification: `gen.py` has overlapping functions with `gen.erl` in OTP, where it contains generic call/response code between `gen_*` everything

File: docs/_sources/calling_python.rst.txt [hunk 3]
```diff
@@ -1,13 +1,15 @@
 Remote Calling Python from Erlang
```
Comment (kvakvs) [inline docs/_sources/calling_python.rst.txt]: This file is same or almost same as `calling_python.rst` what is happening?

File: docs-src/source/data_types.rst (hunk 1 above)
Comment (s2hc-johan) [reply docs-src/source/data_types.rst]: yea but when using generic calls there is a problem with the round trip, but maybe this is can be fixed in a way I don't know about.

In the documentation it's even stated that string() isn't a data type http://erlang.org/doc/reference_manual/data_types.html#string. Char is described under number http://erlang.org/doc/reference_manual/data_types.html#number.

So there exists some friction here I'm just saying we need to look at what works best most of the time.

so when sending something to erlang sending `[104,101,108,108,111]` and not getting `"hello"` or vice versa is an issue. 

We've starting to use it now and it is problematic to get code going with the interpretation that is right now. We can get around it with explicitness, but the idea with Pyrlang was to try and make the interaction as smooth as possible.
an

File: docs/_sources/calling_python.rst.txt (hunk 3 above)
Comment (s2hc-johan) [reply docs/_sources/calling_python.rst.txt]: everything under docs is generated by sphinx, I'm guessing it places the source files there

File: docs-src/source/modules/gen/index.rst (hunk 2 above)
Comment (s2hc-johan) [reply docs-src/source/modules/gen/index.rst]: ok, I'll update

**Pull request merged**